File: src/ARCtrl/JsonIO/LDObject.fs
namespace ARCtrl.Json

open ARCtrl
open ARCtrl.ROCrate
open System
open ARCtrl.ROCrate
open Thoth.Json.Core
open DynamicObj
open Fable.Core

[<AutoOpen>]
module LDNodeExtensions =

    type LDNode with

        static member fromROCrateJsonString (s:string) = 
            Decode.fromJsonString LDNode.decoder s

        /// exports in json-ld format
        static member toROCrateJsonString(?spaces) =
            fun (obj:LDNode) ->
                LDNode.encoder obj
                |> Encode.toJsonString (Encode.defaultSpaces spaces)

        member this.ToROCrateJsonString(?spaces) = 
            LDNode.toROCrateJsonString(?spaces=spaces) this

    [<AttachMembers>]
    type PyJsInterop =

        static member fromROCrateJsonString (s:string) = 
            Decode.fromJsonString LDNode.decoder s

        static member toROCrateJsonString(node : LDNode, ?spaces) =
            LDNode.encoder node
            |> Encode.toJsonString (Encode.defaultSpaces spaces)

[<AutoOpen>]
module LDGraphExtensions =

    type LDGraph with

        static member fromROCrateJsonString (s:string) = 
            Decode.fromJsonString LDGraph.decoder s

        /// exports in json-ld format
        static member toROCrateJsonString(?spaces) =
            fun (obj:LDGraph) ->
                LDGraph.encoder obj
                |> Encode.toJsonString (Encode.defaultSpaces spaces)

        member this.ToROCrateJsonString(?spaces) = 
            LDGraph.toROCrateJsonString(?spaces=spaces) this

    [<AttachMembers>]
    type PyJsInterop =

        static member fromROCrateJsonString (s:string) = 
            Decode.fromJsonString LDGraph.decoder s

        static member toROCrateJsonString(graph : LDGraph, ?spaces) =
            LDGraph.encoder graph
            |> Encode.toJsonString (Encode.defaultSpaces spaces)

File: js/JsonIO/LDObject.js
import { fromString } from "../fable_modules/Thoth.Json.JavaScript.0.3.0/Decode.fs.js";
import { encoder, decoder as decoder_1 } from "../Json/LDObject.js";
import { printf, toText } from "../fable_modules/fable-library-js.4.22.0/String.js";
import { toString } from "../fable_modules/Thoth.Json.JavaScript.0.3.0/Encode.fs.js";
import { defaultSpaces } from "../Json/Encode.js";
import { unwrap } from "../fable_modules/fable-library-js.4.22.0/Option.js";

export function ARCtrl_ROCrate_LDObject__LDObject_fromROCrateJsonString_Static_Z721C83C5(s) {
    const matchValue = fromString(decoder_1, s);
    if (matchValue.tag === 1) {
        throw new Error(toText(printf("Error decoding string: %O"))(matchValue.fields[0]));
    }
    else {
        return matchValue.fields[0];
    }
}

/**
 * exports in json-ld format
 */
export function ARCtrl_ROCrate_LDObject__LDObject_toROCrateJsonString_Static_71136F3F(spaces) {
    return (obj) => {
        const value = encoder(obj);
        return toString(defaultSpaces(spaces), value);
    };
}

export function ARCtrl_ROCrate_LDObject__LDObject_ToROCrateJsonString_71136F3F(this$, spaces) {
    return ARCtrl_ROCrate_LDObject__LDObject_toROCrateJsonString_Static_71136F3F(unwrap(spaces))(this$);
}

